pkg/node/volumemgr.go

```diff
@@ -1226,3 +1234,96 @@ func (m *VolumeManager) overrideDriveHealth(drive *api.Drive, overriddenHealth,
 			driveCRName, overriddenHealth, apiV1.HealthGood, apiV1.HealthSuspect, apiV1.HealthBad, apiV1.HealthUnknown)
 	}
 }
+
+func (m *VolumeManager) setWbtValue(vol *volumecrd.Volume) error {
+	device, err := m.findDeviceName(vol)
+	if err != nil {
+		return err
+	}
+
+	err = m.wbtOps.SetValue(device, m.wbtConfig.Value)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *VolumeManager) restoreWbtValue(vol *volumecrd.Volume) error {
+	device, err := m.findDeviceName(vol)
+	if err != nil {
+		return err
+	}
+
+	err = m.wbtOps.RestoreDefault(device)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *VolumeManager) checkWbtChangingEnable(vol *volumecrd.Volume) bool {
+	if m.wbtConfig == nil {
+		return false
+	}
+
+	if !m.wbtConfig.Enable {
+		return false
+	}
+
+	isModeAcceptable := false
+	for _, mode := range m.wbtConfig.VolumeOptions.Modes {
+		if mode == vol.Spec.Mode {
+			isModeAcceptable = true
+			break
+		}
+	}
+	if !isModeAcceptable {
+		m.log.Infof("Skip wbt value changing: volume %s has mode %s, acceptable: %v", vol.Name, vol.Spec.Mode, m.wbtConfig.VolumeOptions.Modes)
+		return false
+	}
+
+	isSCAcceptable := false
+	for _, sc := range m.wbtConfig.VolumeOptions.StorageTypes {
+		if sc == vol.Spec.StorageClass {
+			isSCAcceptable = true
+			break
+		}
+	}
+	if !isSCAcceptable {
+		m.log.Infof("Skip wbt value changing: volume %s has sc %s, acceptable: %v", vol.Name, vol.Spec.StorageClass, m.wbtConfig.VolumeOptions.StorageTypes)
+		return false
+	}
+
+	return true
+}
+
+func (m *VolumeManager) findDeviceName(vol *volumecrd.Volume) (string, error) {
+	drive, err := m.crHelper.GetDriveCRByVolume(vol)
+	if err != nil {
+		return "", err
+	}
+	if drive == nil {
+		return "", fmt.Errorf("drive %s is not found", vol.Spec.Location)
+	}
+
+	// expected device path - /dev/<device>
+	splitedPath := regexp.MustCompile(`[A-Za-z0-9]+`).FindAllString(drive.Spec.Path, -1)
```

Comment:
Why do we need this check?

Reply:
We must have only device name here without `/dev`